robottelo/cli/factory.py

```diff
@@ -1691,9 +1691,15 @@ def setup_org_for_a_custom_repo(options=None):
     cvv = ContentView.info({'id': cv_id})['versions'][-1]
     # Promote version to next env
     try:
-        ContentView.version_promote(
-            {'id': cvv['id'], 'organization-id': org_id, 'to-lifecycle-environment-id': env_id}
-        )
+        all_lifecycle_env = ContentView.info({'id': cv_id})['lifecycle-environments']
+        for lce in all_lifecycle_env:
+            if int(lce['id']) == env_id:
+                # already promoted to environment
+                break
```

Comment:
why break here, if CV is already promoted, we need to go ahead and follow further steps, right?

Reply:
if no `env_id` matches with `int(lce['id'])` for loop will not break and it will goes inside else to promote at least once.